spec/urn_spec.rb

```diff
@@ -2,51 +2,61 @@
 require 'urn'
 
 RSpec.describe URN do
-  describe '#valid?' do
-    it 'returns true if it is valid' do
-      expect(described_class.new('urn:namespace:specificstring')).to be_valid
+  describe 'Kernel.URN' do
```

Comment:
Might be interesting to look at exactly [how `URI()` works](https://github.com/ruby/ruby/blob/trunk/lib/uri/common.rb#L705-L721); perhaps re-opening `Kernel` is more idiomatic for these things?